main/src/ca/uwaterloo/flix/language/phase/Mutator.scala

```diff
@@ -0,0 +1,149 @@
+package ca.uwaterloo.flix.language.phase
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.language.CompilationMessage
+import ca.uwaterloo.flix.language.ast.Ast.Constant
+import ca.uwaterloo.flix.language.ast.SemanticOp.BoolOp
+import ca.uwaterloo.flix.language.ast.TypedAst._
+import ca.uwaterloo.flix.util.Validation
+import ca.uwaterloo.flix.util.Validation._
+
+object Mutator {
+
+  // TODO: The purpose of this phase is to:
+  // 1. Pick a random def (perhaps rather a def from a specific file ?)
+  // 2. Pick a random expr (how?)
+  // 3. Slightly modify the expr (e.g. changing (x + 1) to (x - 1).
```

Comment:
I don't quite understand why we need random.

Don't we want to implement mutations for various Nodes, which we will use when parsing AST wherever possible?

Reply:
Not sure i understand the question

Reply:
It says here to mutate a random one def/expr. But the rest are all not needed? Or if needed, how to get this with a random selection?

Reply:
What I mean is that when you do mutation testing, you should only mutate one thing at a time.

For example, it would not be sensible to change _every_ `+` to a `-`.

So you have to pick a random `def` and change one if its expressions.